### ISSUE TITLE
Bug/inconsistency when using "days" on Duration vs Period

### ISSUE THREAD

**Issue by RokLenarcic** [juxt/tick#195]
Here's the inconsistency:
```
(t/days (t/between (t/midnight (t/date "2024-02-14")) (t/midnight (t/date "2024-06-13"))))
=> 120
(t/days (t/between (t/date "2024-02-14") (t/date "2024-06-13")))
=> 30
```

So when I use `midnight` to convert LocalDate to LocalDateTime, I get `Duration` instead of `Period`. This behaviour is very surprising, I would say it's bug-like, because you would expect same return for logically same data from same protocol function.

**Comment by henryw374** [comment]
The Period/Duration split in java.time is awkward IMO, so I agree this is far from ideal, but one of the principles of tick is that it behaves as java.time does. 

For example depending on the type of a and b, you'd get a different type from 

`(.until a b) `

The reasoning in java.time is that you should know from the type of a and b what the type the answer should be. 

Having said that, I tend to only use the 3-arity `between` function these days. 

`(t/between a b :days)`

**Comment by henryw374** [comment]
tl;dr I don't think there's anything we should change

**Comment by RokLenarcic** [comment]
It's certainly a position you can take, but from a users perspective it was quite shocking and this produced a serious bug in some tax reports. Dodging this bug requires user to know not just about Period and Duration differences, but also how Tick interacts with them via `days` function and also which type `between` returns. Clojure is dynamically types and this is not at all obvious from APIs. They can play around more with Java due to the type system.
